byway/src/main/webapp/routepage.js

```diff
@@ -247,21 +273,21 @@ function renderRecsList(){
 }
  
 /** Creates a button in the recommendations panel in the html 
- *  @param {String} rec a String to add as a button 
+ *  @param {PlaceResult} rec a PlaceResult object whose name property is added as a button 
  *  @return {button} recBtn a button showing a recommended place
  */
 function createRecButton(rec){
   const recBtn = document.createElement('button');
-  recBtn.innerText = rec;
+  recBtn.innerText = rec.name;
   recBtn.className =  "btn rec-btn";
   recBtn.addEventListener("click", function() {
-    if(!route.includes(rec)){
+    if(!route.some(waypoint => waypoint.name === rec.name)){
       route.push(rec);
       calcRoute();
     }
   });
   return recBtn;
 }
 
-/* exported initMap, generateRoute */
-/* global google */
\ No newline at end of file
+/* exported initMap, generateRoute, placesServices */
+/* global google:true, map:true, findPlace:true*/
```

Comment:
Just listing the globals should be sufficient, where did you get `name:true` from?

Reply:
If I don't put `:true` eslint gives an error saying that global vars should not be modified.

Reply:
`/* global varName */` defaults to treating values like `const`, but can be changed to treat them like `let` with `/* global varName:writeable */`. If you need to make a global writeable, prefer to use `:writeable` rather than `:true` because that is easier to understand.

However: if you are going to be initializing some global variable in a particular file, then you should declare the variable in the same file with `let`, and "import" it using `/* global */` in other files where you only need to use the value, rather than need to assign it. If a particular file handles setting up the value for a global, then it makes sense to also *declare* the global in that file (and mark it as exported). Having declaration and initialization separate is error-prone, since the declaration and assignment are separated its easier to mess up the naming. Also, if you get in the habit of using `:writeable` globals, you may accidentally overwrite globals that you didn't intend to; if you only assign them in the same file where they are declared, then in all other files, the linter will help you make sure that you are not overwriting them by accident.